qiime2/sdk/plugin_manager.py

```diff
@@ -81,15 +81,25 @@ def _init(self, add_plugins):
                 package = entry_point.module_name.split('.')[0]
                 plugin = entry_point.load()
 
-                self.add_plugin(plugin, package, project_name)
+                self.add_plugin(plugin, package, project_name,
+                                consistency_check=False)
 
-    def add_plugin(self, plugin, package=None, project_name=None):
+            self.add_plugin(plugin, package, project_name,
+                            consistency_check=True)
+
+    def _consistency_check(self):
+        pass
+
+    def add_plugin(self, plugin, package=None, project_name=None,
+                   consistency_check=True):
         self.plugins[plugin.name] = plugin
         self._plugin_by_id[plugin.id] = plugin
         if plugin.package is None:
             plugin.package = package
         if plugin.project_name is None:
             plugin.project_name = project_name
+        if consistency_check is True:
+            return self._consistency_check(plugin)
```

Comment:
I would put this after `plugin.freeze()` so that it's the last thing that is run, and there's the maximum amount of time possible to get all of the :duck:s in a row.